src/AqueductV1Pair.sol

```diff
@@ -816,22 +854,26 @@ contract AqueductV1Pair is IAqueductV1Pair, AqueductV1ERC20, SuperAppBase {
         // set user starting cumulative
         (address user, ) = abi.decode(_agreementData, (address, address));
         if (address(_superToken) == _token0) {
-            uint256 balance1 = UQ112x112.decode(
-                uint256(uint96(flow0)) * (twap1CumulativeLast - userStartingCumulatives1[user])
-            );
+            uint256 diff;
+            // underflow desired
+            unchecked {
+                diff = twap1CumulativeLast - userStartingCumulatives1[user];
```

Comment:
Why is this line unchecked when we are subtracting cumulatives, and not similar logic in `retrieveFunds`:
```solidity
returnedBalance = UQ160x96.decode(
    uint256(uint96(flow1)) * (twap0CumulativeLast - userStartingCumulatives0[msg.sender])
);
```

Reply:
resolved in 317239c